tests/python_tests/helpers/golden_generators.py

```diff
@@ -73,12 +229,111 @@ def to_tensor(operand, data_format):
 
 @register_golden
 class MatmulGolden(FidelityMasking):
+
     def __call__(
         self, operand1, operand2, data_format, math_fidelity, input_dimensions=[32, 32]
     ):
         torch_format = format_dict[data_format]
 
-        self._apply_fidelity_masking(operand1, operand2, math_fidelity)
+        t1 = to_tensor(operand1, data_format)
+        t2 = to_tensor(operand2, data_format)
+
+        _fildelity_dict = {
+            MathFidelity.LoFi: 0,
+            MathFidelity.HiFi2: 1,
+            MathFidelity.HiFi3: 2,
+            MathFidelity.HiFi4: 3,
+        }
+
+        num_fidelity_phases = _fildelity_dict.get(math_fidelity, 0)
+
+        res = 0
+
+        if num_fidelity_phases == 0:
+
+            t1, t2 = self._apply_fidelity_masking(t1, t2, 0, data_format)
+            t1, t2 = t1.view(input_dimensions[0], input_dimensions[1]), t2.view(
+                input_dimensions[0], input_dimensions[1]
+            )
+            res = (
+                torch.matmul(t1, t2)
+                .view(input_dimensions[0] * input_dimensions[1])
+                .to(torch_format)
+            )
+
+            return res
+
+        elif num_fidelity_phases == 1:
+
+            t1, t2 = self._apply_fidelity_masking(t1, t2, 0, data_format)
```

Comment:
```suggestion
            t1, t2 = self._apply_fidelity_masking(t1, t2, 1, data_format)
```
Shouldn't this be 1?